### ISSUE TITLE
mongodbatlas_alert_configuration tries to change `notification.interval_min` on every plan

### ISSUE THREAD

**Issue by pedro-te** [mongodb/terraform-provider-mongodbatlas#449]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v0.14.11
+ provider registry.terraform.io/hashicorp/aws v3.37.0
+ provider registry.terraform.io/mongodb/mongodbatlas v0.9.0
```

### Terraform Configuration File 
```hcl
resource "mongodbatlas_alert_configuration" "no-primary" {
  project_id = var.project-id
  event_type = "NO_PRIMARY"
  enabled    = "true"

  notification {
    type_name     = "PAGER_DUTY"
    service_key   = <some_service_key>
    delay_min     = 0
    interval_min  = 5
  }
}
```

### Steps to Reproduce
1. `terraform init`
2. `terraform plan`
3. `terraform apply`
4. `terraform plan` again

### Expected Behavior
No changes seen in the second plan.

### Actual Behavior
```hcl
  # mongodbatlas_alert_configuration.no-primary will be updated in-place
  ~ resource "mongodbatlas_alert_configuration" "no-primary" {
        id                     = "<some_id>"
        # (6 unchanged attributes hidden)

      ~ notification {
          ~ interval_min  = 2147483647 -> 5
            # (6 unchanged attributes hidden)
        }
    }
```

For some reason the value 5 is not stored and instead `2147483647` is in the state. So, terraform will try to set it to 5 on every plan executed. 

### Additional Context
Before forcefully setting `interval_min` to 5, we were not setting it at all. In that scenario, terraform tries to set it to `null` in the second plan:
```hcl
  # mongodbatlas_alert_configuration.no-primary will be updated in-place
  ~ resource "mongodbatlas_alert_configuration" "no-primary" {
        id                     = "<some_id>"
        # (6 unchanged attributes hidden)

      ~ notification {
          - interval_min  = 2147483647 -> null
            # (6 unchanged attributes hidden)
        }
    }
```

**Comment by themantissa** [comment]
@pedro-te thank you for the report.  I'll have the team look into the issue and we'll report back any next steps.

**Comment by leofigy** [comment]
Thanks @pedro-te , right, in that resource, in the creation it only reads the config and send it to the API, and after getting the API response is used to populate the state, so of course there is a missmatch, we need to see if the api is not honoring the attribute value, or if there is a bug in the pointer->value setting :)

**Comment by themantissa** [comment]
Internal ticket INTMDB-204

**Comment by ghost** [comment]
@themantissa  Any update or any ETA on this by when it can be fixed?

With the PagerDuty integration, we don't need to set this value as per the documentation. The integration will take care of this. 

But even with the PagerDuty integration, getting the same issue

**Comment by themantissa** [comment]
@vinodhbasavani No ETA at this time but is in progress.  I'm working internally to get the API itself corrected which will fix it for Terraform.

**Comment by Freyert** [comment]
The API always returns the max 32 bit integer 2,147,483,648 (2^(31-1))? So there's no work around for this?

This is a fairly painful issue for users. Makes plans extremely taxing to review because there are often a lot of alerts. Any updates after 3 months?

**Comment by pedro-te** [comment]
> The API always returns the max 32 bit integer 2,147,483,648 (2^(31-1))? So there's no work around for this?
> 
> This is a fairly painful issue for users. Makes plans extremely taxing to review because there are often a lot of alerts. Any updates after 3 months?

@Freyert I'm not sure if this will help you, but we ended up ignoring changes to that parameter, otherwise we would have the same issue you describe. 

```
  lifecycle {
    ignore_changes = [
      # ignore changes to interval_min because the Atlas API always
      # returns a garbage value for this parameter
      # Internal MongoDB Ticket: INTMDB-204; Support case:<support_Case_id>
      notification.0.interval_min,
    ]
  }
```

**Comment by themantissa** [comment]
@Freyert we continue to work with internal MongoDB teams to get this addressed.  It is on the roadmap for correction.  Thank you for the proposed workaround @pedro-te!

**Comment by themantissa** [comment]
Internal ticket to update Terraform to corrected API resource INTMDB-272

**Comment by themantissa** [comment]
This was addressed in version 1.2.0 of the provider.  Thank you!
